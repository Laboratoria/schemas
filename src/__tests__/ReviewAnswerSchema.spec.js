const mongoose = require('mongoose/browser');
const { ReviewAnswerSchema, ReviewQuestionSchema, ProjectFeedbackSchema } = require('../../')(mongoose);

describe('ReviewAnswerSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, ReviewAnswerSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should pass sync validation ...', () => {
    const question = new mongoose.Document({}, ReviewQuestionSchema);
    const projectFeedback = new mongoose.Document({}, ProjectFeedbackSchema);
    const doc = new mongoose.Document({
      question: question._id,
      projectFeedback: projectFeedback._id,
      value: 'Crack',
    }, ReviewAnswerSchema);
    expect(doc.validateSync()).toBe(undefined);
  });

  it('should pass validation when answer is missing value', () => {
    const question = new mongoose.Document({}, ReviewQuestionSchema);
    const projectFeedback = new mongoose.Document({}, ProjectFeedbackSchema);
    const doc = new mongoose.Document({
      question: question._id,
      projectFeedback: projectFeedback._id,
      value: 'Crack',
    }, ReviewAnswerSchema);

    return doc.validate();
  });
});
