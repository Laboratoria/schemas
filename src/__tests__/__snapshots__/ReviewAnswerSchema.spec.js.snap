// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewAnswerSchema should fail validation when fields missing 1`] = `
Object {
  "projectFeedback": [ValidatorError: Path \`projectFeedback\` is required.],
  "question": [ValidatorError: Path \`question\` is required.],
}
`;
